### PR TITLE
[Part 1] Implementation for Exclusion List in AppTP

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/0/1203134519344743/f

### Description
Removed the last screen in the AppTP onboarding and added a Dax dialog.

### Steps to test this PR

- [x] Fresh install from this branch.
- [x] Enable AppTP from Settings and pay attention to the onboarding flow.
- [x] After you grant the VPN permission: 
    - [x] Check that you don't see the "Hooray, you're all set up" screen anymore. 
    - [x] Check that you see the current "Recent activity" and a Dax dialog appears and it looks as in designs.
    - [x] Check that once you tap on the Dax dialog button you see the confetti on screen and the `m_atp_ev_apptp_enabled_cta_button_press` pixel is sent.

### UI changes
apptp_onboarding_dax

### CODE REVIEW

File: app-tracking-protection/vpn-impl/src/main/res/values/strings-vpn.xml [hunk 1]
```diff
@@ -207,7 +208,22 @@
     <string name="atp_ActivityTrackersCompanyBlockedOtherTimesOneCompany"><![CDATA[%1$d tracking attempts blocked from %2$d company in <b>%3$s</b> app]]></string>
     <string name="atp_ActivityTrackersCompanyBlockedOtherTimesOtherCompanies"><![CDATA[%1$d tracking attempts blocked from %2$d companies in <b>%3$s</b> app]]></string>
 
+    <!-- TODO [ANA] New strings added, need translation -->
```
Comment (anikiki) [inline app-tracking-protection/vpn-impl/src/main/res/values/strings-vpn.xml:211]: Please ignore the strings and TODOs. They are just a marker for me and will be cleaned up in the last PR.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt [hunk 2]
```diff
@@ -499,10 +517,69 @@ class DeviceShieldTrackerActivity :
...
+            override fun onDaxDialogSecondaryCtaClick() {
+                // NO OP
+            }
+
+            override fun onDaxDialogHideClick() {
+                // NO OP
+            }
+        })
+
+        handler.postDelayed(delayInMillis = APPTP_ENABLED_CTA_DIALOG_DELAY_MS) {
```
Comment (malmstein) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt:551]: what's the reason for this?

**Review by malmstein — COMMENTED** [review]
Looks good, couple comments on the handler.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/pixels/DeviceShieldPixelNames.kt [hunk 3]
```diff
@@ -230,5 +230,7 @@ enum class DeviceShieldPixelNames(override val pixelName: String, val enqueue: B
     ATP_REPORT_UNPROTECTED_APPS_BUCKET("m_atp_unprotected_apps_bucket_%d_c"),
     ATP_REPORT_UNPROTECTED_APPS_BUCKET_DAILY("m_atp_unprotected_apps_bucket_%d_d"),
 
+    ATP_DID_PRESS_APPTP_ENABLED_CTA_BUTTON("m_atp_ev_apptp_enabled_cta_button_press"),
```
[the remark this answers is not in the document]
Comment (anikiki) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/pixels/DeviceShieldPixelNames.kt:233]: If everything works as expected this pixel is sent only once when the button on the dialog is pressed. The dialog is not shown again unless the app is uninstalled / storage cleared.

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt (hunk 2 above)
Comment (anikiki) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt:551]: The handler will be removed in the last PR.

**Review by malmstein — APPROVED** [review]
Works as expected. Do you want to add a TODO for the `Handler `? As long as it's not there later it's up to you how you handle it :)

**Comment by anikiki** [comment]
> Works as expected. Do you want to add a TODO for the `Handler `? As long as it's not there later it's up to you how you handle it :)

Thanks, added :+1:

**Pull request closed without merging**